src/systems/material.js

```diff
@@ -146,7 +146,7 @@ module.exports.System = registerSystem('material', {
     setTextureProperties(texture, data);
 
     // if we're on iOS, and the video is HLS, we currently need to do some hacks
-    if (utils.device.isIOS() && isHLS(videoEl)) {
+    if (utils.device.isIOS() && isHLS(videoEl.src || videoEl.getAttribute('src'), videoEl.type || videoEl.getAttribute('type'))) {
```

Comment:
Just to confirm, you tried changing `utils.device.isIOS()` here to `this.el.sceneEl.isIOS`, and in the test, try mocking `sceneEl.isIOS = true` before the assertions?